singer_sdk/target_base.py

```diff
@@ -530,10 +530,13 @@ def _write_state_message(self, state: dict) -> None:
         Args:
             state: TODO
         """
-        state_json = json.dumps(state)
-        self.logger.info("Emitting completed target state %s", state_json)
-        sys.stdout.write(f"{state_json}\n")
-        sys.stdout.flush()
+        if state:
```

Comment:
We could also make this change on line 493

```diff
- if self._latest_state is not None:
+ if self._latest_state:
```

would that work?

Reply:
Oh, my PR was outdated! 🤦 I didn't see this change! Yep, I think `if self._latest_state:` should work. Let me change it.

Reply:
Changed in https://github.com/meltano/sdk/pull/3040/commits/bfb6d84546ea4a3f611e3f8c8b37da05fc5c8f6c